### ISSUE TITLE
softmaxlog implementation in autonn

### ISSUE THREAD

**Issue by ncalik** [vlfeat/autonn#45]
Hi @jotaf98, I wanna implement a new loss function in autonn. 
Firstly, i've tried to write softmaxlog like : 

xc = vl_nnconv(xc,'size', [1, 1, 4096, 2], 'stride', 1,'pad',0);
lbl = Input();
 
Xmax       = max(xc ,[],3);
tmp_vect = xc - Xmax;
exp_vect  = exp(tmp_vect);
obj_sum   =   Xmax  +    log(sum(exp_vect,3))  -  xc(1,1,lbl); 

but, i didn't get same results with vl_nnloss softmaxlog. How can i write in autonn?

**Comment by ncalik** [comment]
I also tried this : 

 obj_sum = 0;
 for i = 1:numel(lbl)
     tmp_vect = xc(1,1,:,i);
     Xmax     = max(tmp_vect,[],3);
     tmp_vect = tmp_vect - Xmax;
     exp_vect = exp(tmp_vect);
     obj_sum  =   Xmax  +    log(sum(exp_vect,3)) - xc(1,1,lbl(i),i);
 end
 obj_sum = obj_sum/numel(lbl);
